holdings_maintenance/pds4/pds4linkshelf.py

```
@@ -0,0 +1,1270 @@
+    merged_limits = VALIDATE_LINKS_LIMITS.copy()
+    merged_limits.update(limits)
+    logger.open('Validating link shelf file for', dirpath, limits=merged_limits)
+
+    try:
+        keys = list(dirdict.keys())
+        for key in keys:
+            if key in shelfdict:
+                dirinfo = dirdict[key]
+                shelfinfo = shelfdict[key]
+
+                if type(dirinfo) == list:
```

Flake8 (7.3.0)

[error] do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`

(E721)